File: backend/domain/src/main/java/kr/co/jparangdev/boardbuddy/domain/game/Game.java
package kr.co.jparangdev.boardbuddy.domain.game;

import java.time.Instant;

import lombok.*;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Game {
    private Long id;
    private String name;
    private String nameKo;
    private String nameEn;
    private int minPlayers;
    private int maxPlayers;
    private ScoreStrategy scoreStrategy;
    private Instant createdAt;

    @Builder
    public Game(Long id, String name, String nameKo, String nameEn, int minPlayers, int maxPlayers,
                ScoreStrategy scoreStrategy, Instant createdAt) {
        this.id = id;
        this.name = name;
        this.nameKo = nameKo;
        this.nameEn = nameEn;
        this.minPlayers = minPlayers;
        this.maxPlayers = maxPlayers;
        this.scoreStrategy = scoreStrategy;
        this.createdAt = createdAt;
    }

    public static Game create(String name, String nameKo, String nameEn, int minPlayers, int maxPlayers, ScoreStrategy scoreStrategy) {
        return Game.builder()
                .name(name)
                .nameKo(nameKo)
                .nameEn(nameEn)
                .minPlayers(minPlayers)
                .maxPlayers(maxPlayers)
                .scoreStrategy(scoreStrategy)
                .createdAt(Instant.now())
                .build();
    }

    public void update(String nameKo, String nameEn) {
        this.nameKo = nameKo;
        this.nameEn = nameEn;
    }
}

File: frontend/src/utils/game.ts
import type {CustomGame, Game} from '@/types';

/**
 * 현재 언어에 맞는 게임 이름을 반환합니다.
 * 우선순위: 해당 언어 이름 > 기본 name
 */
export function getGameName(game: Game | CustomGame, locale: string): string {
  if (locale === 'ko' && game.nameKo) {
    return game.nameKo;
  }
  if (locale === 'en' && game.nameEn) {
    return game.nameEn;
  }
  return game.name;
}

/**
 * 검색을 위해 모든 가능한 게임 이름을 반환합니다.
 */
export function getAllGameNames(game: Game | CustomGame): string[] {
  const names = [game.name];
  if (game.nameKo) names.push(game.nameKo);
  if (game.nameEn) names.push(game.nameEn);
  return names;
}
